src/index.ts

```diff
@@ -6,3 +6,11 @@ export * from './components';
 
 export {UnstyledLinkProps, LinkLikeComponent} from './utilities/link';
 export {createPolarisContext} from './utilities/create-polaris-context';
+export {
+  ScrollLockManagerContext as __UNSAFE_SECRET_INTERNAL_SCROLL_LOCK_MANAGER_CONTEXT,
+} from './utilities/scroll-lock-manager';
+export {
+  WithinContentContext as __UNSAFE_SECRET_INTERNAL_WITHIN_CONTENT_CONTEXT,
```

Comment:
We shouldn't really expose `contexts`. There's a private part of the app, and having this publicly available could harm us in the future. We shouldn't be promoting the use of it. From a design perspective if we decide that `WithinContext` is a variant that could be opted into we should probably expose a prop, rather than our internals. However keeping the export here will allow it to be used, if needed (which was by forked components in web).

Reply:
As an alternative name we could do with `__UNSTABLE_INTERNAL_WITHIN_CONTENT_CONTEXT`. And we should mention that `_UNSTABLE` is to be used at the consumer's own risk and may receive breaking changes in minor versions

Reply:
On second thoughts keeping this scarey means its' not strictly speaking public API and we can change it up in a minor version. Let's keep it this way;

Reply:
(Totally missed your replies when I commented there, and it sounds like I ignored ya, sorry)

You point about not exposing contexts is a good one. Keeping this scary internal name now gives us a chance to come up with a proper design for this later.


Poking about with naming - unstable and unsafe suggest they might become stable later, but it sounds like we might want to go a different route with how we expose contexts. [React has __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED](https://github.com/facebook/react/blob/9b0bd43550206e04bfe9ca695e5981eff0e2d03f/packages/react/src/React.js#L104), so perhaps drop that part of the prefix and have `__SECRET_INTERNAL_SCROLL_LOCK_MANAGER_CONTEXT` and `__SECRET_INTERNAL_WITHIN_CONTENT_CONTEXT`?